### PR TITLE
HP-2603 Feat/manifest filename

### PR DESCRIPTION
**Pull request by mfshao**
Link to JIRA ticket if there is one: https://ctds-planx.atlassian.net/browse/HP-2603

### Improvements
- Discovery: downloaded file manifests now will have hostname, identifier and date information in their filenames
- Discovery: multiple file manifests generated for downloading will be put into a single archive file

### CODE REVIEW

File: src/Discovery/DiscoveryDetails/Components/FieldGrouping/TabField/DataDownloadList/ActionButtons/DownloadUtils/GenerateFilename.test.ts [hunk 1]
```diff
@@ -71,6 +71,12 @@ describe('GenerateFilename', () => {
       );
     });
 
+    test('omits studyID segment for "all_manifests"', () => {
```
Comment (jarvisraymond-uchicago) [inline src/Discovery/DiscoveryDetails/Components/FieldGrouping/TabField/DataDownloadList/ActionButtons/DownloadUtils/GenerateFilename.test.ts:73]: FYI, this file is failing ESLINT with error: 
`  1:8  error  'moment' is defined but never used  @typescript-eslint/no-unused-vars
`
This is because moment is imported but not used, just mocked. 

ESLINT can be run locally with
`npm run eslint`
or
`npm run eslint-new` 
to test only files that differ from master

File: src/Discovery/DiscoveryActionBar/utils/handleDownloadManifestClick.ts [hunk 2]
```diff
@@ -16,11 +18,17 @@ const handleDownloadManifestClick = (
...
+
   if (healIDPLoginNeeded) {
     return;
   }
-  // combine manifests from all selected studies
-  const { fileManifest } = assembleFileMetadata(manifestFieldName, selectedResources);
+
+  const fileManifestsWithNames = processFileMetadata(uidFieldName, manifestFieldName, selectedResources);
+  if (!Object.keys(fileManifestsWithNames).length) {
+    return;
```
Comment (jarvisraymond-uchicago) [inline src/Discovery/DiscoveryActionBar/utils/handleDownloadManifestClick.ts:29]: Question: in this situation would be want to show an error to the end user? Also the same with line 23. 

If this branch executes, clicking the button causes nothing to happen and the UI doesn't alert the user that anything is wrong. 

Image

**Review by jarvisraymond-uchicago — COMMENTED** [review]
Overall code looks good! Functionality works as expected, identified an edge case we might want to address and a lint error. 

* Confirmed downloading single manifest files results in file name format specified in ticket
* Confirmed downloading multiple manifest files results in file name format specified in ticket
* Confirmed updates to JEST tests pass
* Noted ESLINT issue in `src/Discovery/DiscoveryDetails/Components/FieldGrouping/TabField/DataDownloadList/ActionButtons/DownloadUtils/GenerateFilename.test.ts`
* Noted confusing UX edge case in `src/Discovery/DiscoveryActionBar/utils/handleDownloadManifestClick.ts`

File: src/Discovery/DiscoveryActionBar/utils/handleDownloadManifestClick.ts (hunk 2 above)
Comment (mfshao) [reply src/Discovery/DiscoveryActionBar/utils/handleDownloadManifestClick.ts:29]: I don't think so, because this is a check ofor the component that conceptually should not be triggered, becuase in the actual download button component there is another guard condition https://github.com/uc-cdis/data-portal/blob/master/src/Discovery/DiscoveryActionBar/components/DownloadManifestButton.tsx#L71, which should just disable the download button if there is no manifests in selected studies

**Review by jarvisraymond-uchicago — APPROVED** [review]
